src/pages/api/account/refresh-tag.js

```diff
@@ -0,0 +1,36 @@
+import HTTP_CODE from "@/lib/constants/httpStatusCodes";
+import { makeBanner } from "@/lib/riitag/banner";
+import logger from "@/lib/logger";
+import { ncWithSession } from "@/lib/routing";
+
+async function refreshTag(request, response) {
+    const username = request.session?.username;
+
+    if (!username) {
+        return response
+        .status(HTTP_CODE.UNAUTHORIZED)
+        .json({ error: 'Unauthorized' });
+    }
+
+    const user = await prisma.user.findFirst({
+        where: {
+            username,
+        }
+    });
+
+    try {
+        await makeBanner(user);
+    } catch (error) {
+        logger.error(error);
+        return response
+            .status(HTTP_CODE.INTERNAL_SERVER_ERROR)
+            .json({ error: "Error while creating tag." 
+        });
+    }
+
+    return response.status(HTTP_CODE.OK).json({ success: true });
+}
+
+const handler = ncWithSession().get(refreshTag);
```

Comment:
Use `post` to avoid CSRF

Reply:
I'm unsure what you mean here?

Reply:
You should change the `get` to `post` together with "`method:`" where this URL is called (I commented on it in the other file). This is so that an attacker can't just do actions on behalf of the user.